docs/application/dotnet/guides/nui/visuals.md

```diff
@@ -1,390 +1,552 @@
+   _borderVisual.SizePolicy = VisualTransformPolicyType.Relative;
+   // the visual size (width and height) are equal half of the parent size             
+   _borderVisual.RelativeSize = new RelativeVector2(0.5f, 0.5f);
+   // position interpreted in absolute units
+   _borderVisual.PositionPolicy = VisualTransformPolicyType.Absolute;
+   // the reference point of the Visual is shifted right (50 pixels) and up (100 pixels) from the parent's reference point
+   _borderVisual.Position = new Vector2(50.0f, -100.0f);
+   // the parent's reference point
+   _borderVisual.Origin = Visual.AlignType.BottomBegin;
+   // point on the parent with respect to which the Visual will be placed
+   _borderVisual.AnchorPoint = Visual.AlignType.BottomBegin;
 
-The following example illustrates how to use a `BorderVisual` visual map. The visual is created with the `AddVisual()` method.
+   // _visualView is a previously created VisualView
+   _visualView.AddVisual("nameOfBorderVisual", _borderVisual);
+   ```
 
-```
-private BorderVisual borderVisualMap1;
 
-borderVisualMap1 = new BorderVisual();
 
-borderVisualMap1.Color = Color.Red;
-borderVisualMap1.BorderSize = 5.0f;
+**Figure: BorderVisual obtained from the above code (_visualView background color set to gray).**
 
-borderVisualMap1.Size = new Vector2(100.0f, 100.0f);
-borderVisualMap1.Position = new Vector2(10.0f, 380.0f);
-borderVisualMap1.PositionPolicy = VisualTransformPolicyType.Absolute;
-borderVisualMap1.SizePolicy = VisualTransformPolicyType.Absolute;
-borderVisualMap1.Origin = Visual.AlignType.TopBegin;
-borderVisualMap1.AnchorPoint = Visual.AlignType.TopBegin;
+![BorderVisual](media/visuals/BorderVisualExample.png)
 
-_visualView.AddVisual("borderVisual1", borderVisualMap1);
-```
 
-<a name="colorvisual"></a>
-## Color Visual
 
-The color visual renders a solid color to the control's quad.
+<!---------------------------------------------------------------------------------------------------------------------------->
+<a name="colorVisual"></a>
+### ColorVisual
 
-**Figure: Color visual**
+The ColorVisual renders a solid rectangle.
 
-![Color visual](media/color-visual.png)
 
-The following table lists the supported properties. The visual type is `Color` and the visual map is `ColorVisual`.
+The properties specific for a ColorVisual are presented in the below table:
+
+**Table: ColorVisual properties**
+
+| Property              | Type    | Required | Description                                                                                |
+|-----------------------|---------|----------|--------------------------------------------------------------------------------------------|
+| `Color`               | `Color` | Yes      | The color of the Visual.                                                                   |
+| `RenderIfTransparent` | `bool`  | No       | Specifies whether to render in case of the `MixColor` being transparent (default `false`). |
```

Comment:
```suggestion 
| `RenderIfTransparent` | `bool`  | No       | Specifies whether to render in case of the `MixColor` being transparent (default `false`). |
```

Whether to render What ? 
This seems incomplete

Reply:
the visual:
Specifies whether to render the visual in case of the `MixColor` being transparent (...)